### PR TITLE
Allow internal clients to set / change the "inCall" flags.

### LINKED ISSUE #420: Allow to customize "inCall" flag for internal clients
Currently [the `inCall` flag of internal clients is always set to `FlagInCall | FlagWithAudio`](https://github.com/strukturag/nextcloud-spreed-signaling/blob/bb24bf5f0d8d787d2776f9873b7b39ea4d0ab785/room.go#L574). Some internal clients may need different flags (for example, a recording server would be a listener/watcher and therefore only need `FlagInCall`), so internal clients should be able to adjust those flags as needed.

### PR DESCRIPTION
**Pull request by fancycode**
Resolves #420 

@danxuliu FYI

### CODE REVIEW

File: docs/standalone-signaling-api-v1.md [hunk 1]
```diff
@@ -869,6 +871,98 @@ Message format (Server -> Client):
...
+
+    {
+      "type": "internal",
+      "internal": {
+        "type": "updatesession",
+        "updatesession": {
+          "sessionid": "the-virtual-sessionid",
+          "roomid": "the-room-id-to-add-the-session",
+          "flags": "optional-initial-flags",
+          "incall": "optional-initial-incall"
```
Comment (danxuliu) [inline docs/standalone-signaling-api-v1.md:928]: Should this be `optional-initial-flags` -> `optional-updated-flags` and `optional-initial-incall` -> `optional-updated-incall` instead?

File: docs/standalone-signaling-api-v1.md [hunk 2]
```diff
@@ -869,6 +871,98 @@ Message format (Server -> Client):
...
+sending media without a regular client connected. This is used for example by
+the SIP bridge to publish mixed phone audio and show "virtual" sessions for the
+individial phone calls.
+
+See above for details on how to connect as internal client. By default, internal
+clients have their "inCall" and the "publishing audio" flags set. Virtual
+sessions have their "inCall" and the "publishing phone" flags set.
+
+This can be changed by including the client feature flag `internal-incall`
+which will require the client to set the flags as necessary.
```
Comment (danxuliu) [inline docs/standalone-signaling-api-v1.md:887]: The flags need to be set in an explicit `incall` message, and they will be 0 until then, right? Or is it possible to set them already in the `hello` message?

**Review by danxuliu — COMMENTED** [review]
Tested the `incall` message in https://github.com/nextcloud/spreed/pull/8756 and works as expected :-) I have not tested the other messages, though.

File: docs/standalone-signaling-api-v1.md (hunk 2 above)
Comment (fancycode) [reply docs/standalone-signaling-api-v1.md:887]: They are initially `0` and will need to be set with an `incall` message. I'm rather against including all kinds of optional parameters to the `hello` message.

**Comment by fancycode** [comment]
Thanks for the testing and feedback! The other messages are already used by the SIP bridge, I just added documentation for them while I was at it.

File: docs/standalone-signaling-api-v1.md (hunk 2 above)
Comment (danxuliu) [reply docs/standalone-signaling-api-v1.md:887]: > I'm rather against including all kinds of optional parameters to the hello message.

Totally fine, I just wanted to be sure that I was getting it right :-)

**Pull request merged**